src/ILLink.RoslynAnalyzer/DataFlow/LocalDataFlowVisitor.cs

```diff
@@ -176,6 +176,10 @@ public override TValue VisitSimpleAssignment (ISimpleAssignmentOperation operati
 				// This can happen for a field assignment in an attribute instance.
 				// TODO: validate against the field attributes.
 				break;
+			case IInstanceReferenceOperation:
+				// Assignment to 'this' is not tracked currently.
+				// Not relevant for trimming dataflow.
```

Comment:
Technically if the method itself is annotated with DAM - that annotation actually applies to `this` and so if the method assigned to this, the source should have matching annotations.

That said - what is the code which would even do this in the first place?
And we only allow annotating methods on corelib Type related types, so it doesn't matter if we don't do anything.